UefiCpuPkg/PiSmmCpuDxeSmm/PiSmmCpuDxeSmm.c

```diff
@@ -391,3 +391,43 @@ PiCpuSmmEntry (
 
   return Status;
 }
+
+/**
+  Initialize SMM profile in SmmReadyToLock protocol callback function.
+
+  @param  Protocol   Points to the protocol's unique identifier.
+  @param  Interface  Points to the interface instance.
+  @param  Handle     The handle on which the interface was installed.
+
+  @retval EFI_SUCCESS SmmReadyToLock protocol callback runs successfully.
+**/
+EFI_STATUS
+EFIAPI
+InitSmmProfileCallBack (
+  IN CONST EFI_GUID  *Protocol,
+  IN VOID            *Interface,
+  IN EFI_HANDLE      Handle
+  )
+{
+  EFI_STATUS                 Status;
+  EFI_SMM_VARIABLE_PROTOCOL  *SmmProfileVariable;

```

Comment:
SmmProfileVariable->SmmVariable

Reply:
Can you still implement the InitSmmProfileCallback() in SmmProfile.c and only perform the variable writing when mIsStandaloneMm is FALSE?
This can eliminate two instances of InitSmmProfileCallback().

Reply:
Thanks, will do that.